src/pkg/fault/fault_test.go

```diff
@@ -0,0 +1,202 @@
+package fault_test
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/alcionai/corso/src/pkg/fault"
+)
+
+type FaultErrorsUnitSuite struct {
+	suite.Suite
+}
+
+func TestFaultErrorsUnitSuite(t *testing.T) {
+	suite.Run(t, new(FaultErrorsUnitSuite))
+}
+
+func (suite *FaultErrorsUnitSuite) TestNew() {
+	t := suite.T()
+
+	n := fault.New(false)
+	assert.NotNil(t, n)
+
+	n = fault.New(true)
+	assert.NotNil(t, n)
+}
+
+func (suite *FaultErrorsUnitSuite) TestErr() {
+	table := []struct {
+		name     string
+		failFast bool
+		fail     error
+		add      error
+		expect   assert.ErrorAssertionFunc
+	}{
+		{
+			name:   "nil",
+			expect: assert.NoError,
+		},
+		{
+			name:     "nil, failFast",
+			failFast: true,
+			expect:   assert.NoError,
+		},
+		{
+			name:   "failed",
+			fail:   assert.AnError,
+			expect: assert.Error,
+		},
+		{
+			name:     "failed, failFast",
+			fail:     assert.AnError,
+			failFast: true,
+			expect:   assert.Error,
+		},
+		{
+			name:   "added",
+			add:    assert.AnError,
+			expect: assert.NoError,
+		},
+		{
+			name:     "added, failFast",
+			add:      assert.AnError,
+			failFast: true,
+			expect:   assert.Error,
+		},
+	}
+	for _, test := range table {
+		suite.T().Run(test.name, func(t *testing.T) {
+			n := fault.New(test.failFast)
+			require.NotNil(t, n)
+
+			e := n.Fail(test.fail)
+			require.NotNil(t, e)
+
+			e = n.Add(test.add)
+			require.NotNil(t, e)
+
+			test.expect(t, n.Err())
+		})
+	}
+}
+
+func (suite *FaultErrorsUnitSuite) TestFail() {
+	t := suite.T()
+
+	n := fault.New(false)
+	require.NotNil(t, n)
+
+	n.Fail(assert.AnError)
+	assert.Error(t, n.Err())
+	assert.Empty(t, n.Errs())
+
+	n.Fail(assert.AnError)
+	assert.Error(t, n.Err())
+	assert.NotEmpty(t, n.Errs())
+}
+
+func (suite *FaultErrorsUnitSuite) TestErrs() {
+	table := []struct {
+		name     string
+		failFast bool
+		fail     error
+		add      error
+		expect   assert.ValueAssertionFunc
+	}{
+		{
+			name:   "nil",
+			expect: assert.Empty,
+		},
+		{
+			name:     "nil, failFast",
+			failFast: true,
+			expect:   assert.Empty,
+		},
+		{
+			name:   "failed",
+			fail:   assert.AnError,
+			expect: assert.Empty,
+		},
+		{
+			name:     "failed, failFast",
+			fail:     assert.AnError,
+			failFast: true,
+			expect:   assert.Empty,
+		},
+		{
+			name:   "added",
+			add:    assert.AnError,
+			expect: assert.NotEmpty,
+		},
+		{
+			name:     "added, failFast",
+			add:      assert.AnError,
+			failFast: true,
+			expect:   assert.NotEmpty,
+		},
+	}
+	for _, test := range table {
+		suite.T().Run(test.name, func(t *testing.T) {
+			n := fault.New(test.failFast)
+			require.NotNil(t, n)
+
+			e := n.Fail(test.fail)
+			require.NotNil(t, e)
+
+			e = n.Add(test.add)
+			require.NotNil(t, e)
+
+			test.expect(t, n.Errs())
+		})
+	}
+}
+
+func (suite *FaultErrorsUnitSuite) TestAdd() {
```

Comment:
nit: I would say this is testing adding multiple times since previous tests already check `Add` to some extent. Same with the `Fail` test

Reply:
True, there's overlap here.  But the difference between Add and Multiple Adds is not, I think, worth calling out.